### PR TITLE
feat(borrowing_service): implement list, retrieve, create, return endpoints

### PR DESCRIPTION
**Pull request by VladKyianytsia**

### CODE REVIEW

File: borrowing_service/models.py [hunk 1]
```diff
@@ -6,12 +8,13 @@
 
 
 class Borrowing(models.Model):
-    borrow_date = models.DateField(auto_now_add=True,)
+    borrow_date = models.DateField()
```
Comment (MaxymChyncha) [inline borrowing_service/models.py]: Why did you remove `auto_now_add` parameter? We need to have this field as autofilled

Comment (VladKyianytsia) [reply borrowing_service/models.py]: Cause i need `borrow_date` in validation `borrow_date > expected_borrow_date`, so i add borrow_date during creation

File: borrowing_service/models.py [hunk 2]
```diff
@@ -25,15 +28,37 @@ class Meta:
...
+        next_borrowing = Borrowing.objects.filter(
+            book_id=book_id,
+            expected_return_date__gt=current_date,
+        ).order_by("expected_return_date").first()
 
-        if self.book.inventory == 0:
+        if next_borrowing:
+            return next_borrowing.expected_return_date
+
+        return None
```
Comment (MaxymChyncha) [inline borrowing_service/models.py]: It seems to me that you don't need to write this statement

File: borrowing_service/serializers.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        with transaction.atomic():
+            book = validated_data["book"]
+            Borrowing.validate_book_inventory(
+                book.inventory,
+                book.title,
+                book.id,
+            )
+            book.inventory -= 1
+            book.save()
+            validated_data["borrow_date"] = datetime.date.today()
```
Comment (MaxymChyncha) [inline borrowing_service/serializers.py]: Isn't it the same if you add `auto_now_add` parameter to your model?

File: borrowing_service/views.py [hunk 4]
```diff
@@ -1,3 +1,69 @@
...
+    viewsets.GenericViewSet,
+):
+    queryset = Borrowing.objects.all()
+    serializer_class = BorrowingSerializer
+    permission_classes = [IsAuthenticated,]
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if is_active := self.request.query_params.get("is_active", False):
```
Comment (MaxymChyncha) [inline borrowing_service/views.py]: Do you need to use walrus operator here if you don't use this variable?

File: borrowing_service/views.py [hunk 5]
```diff
@@ -1,3 +1,69 @@
...
+            queryset = queryset.filter(actual_return_date__isnull=True)
+
+        if self.request.user.is_staff:
+
+            if user_id := self.request.query_params.get("user_id"):
+                queryset = queryset.filter(user_id=user_id)
```
Comment (MaxymChyncha) [inline borrowing_service/views.py:35]: You no need to use this logic. You can get user from your request and make queryset filter with it.

**Review by oksankakot — APPROVED** [review]
